### PR TITLE
fix vulnerabilities (express@4.16.3), rewrite new Buffer() usages

### PR DESCRIPTION
**Pull request by Farfurix**
Before:
```
found 46 vulnerabilities (13 low, 16 moderate, 16 high, 1 critical)
```
Current:
```
found 30 vulnerabilities (8 low, 11 moderate, 10 high, 1 critical)
```

### Changes:
1. Update express.
2. Rewrite `new Buffer()` usages [(deprecated since: v6.0.0)].(https://nodejs.org/api/buffer.html#buffer_new_buffer_array)

>`DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`

3. Add promisify-event on ws.close() in WebSocket server test to avoid Error [ERR_STREAM_DESTROYED]:

> `Error: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed`
>   `1) "after each" hook`

4. Fix playground after `express` update.

### `npm install --save-dev express@4.16.3`
https://github.com/expressjs/express/wiki/Migrating-from-3.x-to-4.x#rescharset:

> `res.charset`
> If you want Express to set a default charset (and you should!), use `res.set('content-type')` or `res.type()` to set the header.
> 
> A default charset will NOT be added when using `res.setHeader()`.

https://github.com/expressjs/express/wiki/Migrating-from-3.x-to-4.x#ressetheaderset-cookie-val:

> `res.setHeader('Set-Cookie', val)`
> This will no longer implicitly append val to the current list of `Set-Cookie` values. You will want to do that manually or use the `res.cookie` method to set cookies, which does this.

### Fix playground
https://expressjs.com/en/api.html#req.body:

> Contains key-value pairs of data submitted in the request body. By default, it is undefined, and is populated when you use body-parsing middleware such as body-parser and multer.

http://expressjs.com/en/api.html#res.sendFile:
> express deprecated res.sendfile: Use res.sendFile instead

### CODE REVIEW

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

File: test/server/proxy-test.js [hunk 1]
```diff
@@ -1845,7 +1848,11 @@ describe('Proxy', () => {
                 .then(msg => {
                     expect(msg).eql('echo');
 
+                    const wsClosePromise = promisifyEvent(ws, 'close');
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `const wsCloseEventPromise =...`

File: test/server/proxy-test.js [hunk 2]
```diff
@@ -2829,7 +2840,7 @@ describe('Proxy', () => {
                 port:     2000,
                 path:     '/wait/150',
                 method:   'GET',
-                body:     new Buffer([]),
+                body:     Buffer.from([]),
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `Buffer.alloc(0)`

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot \retest

**Pull request merged**